File: testdata/go/default/nested_declaration_forms.go
package sample

func nestedDeclarationForms(nums []int, in <-chan int, v interface{}) int {
	total := 0

	if n := len(nums); n > 0 {
		total += n
	}

	for i := 0; i < len(nums); i++ {
		total += nums[i]
	}

	switch mode := total % 2; mode {
	case 0:
		total += mode
	default:
		total += 1
	}

	switch x := v.(type) {
	case int:
		total += x
	case interface{ value() int }:
		total += x.value()
	}

	for idx, val := range nums {
		total += idx + val
	}

	select {
	case n := <-in:
		total += n
	default:
		total += 0
	}

	select {
	case n, ok := <-in:
		if ok {
			total += n
		}
	default:
		total += 0
	}

	return total
}

File: testdata/typescript/default/nested_declaration_forms.ts
function nestedDeclarationForms(items: number[], input: unknown): number {
  let total = 0;

  if (items.length > 0) {
    const count = items.length;
    total += count;
  }

  for (let i = 0; i < items.length; i++) {
    total += items[i];
  }

  for (const [idx, value] of items.entries()) {
    total += idx + value;
  }

  const pair = { left: 1, right: 2 };
  const { left, right: renamedRight } = pair;
  total += left + renamedRight;

  const apply = (value: number): number => {
    const doubled = value * 2;
    return doubled;
  };

  total += apply(total);

  try {
    if (typeof input === "number") {
      total += input;
    }
  } catch (error) {
    total += String(error).length;
  }

  return total;
}
